File: app/src/main/java/edu/gatech/spellastory/story_template_list/StoryListAdapter.java
package edu.gatech.spellastory.story_template_list;

import android.content.Context;
import android.content.Intent;
import android.graphics.drawable.Drawable;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.TextView;

import java.io.IOException;
import java.io.InputStream;
import java.util.List;

import edu.gatech.spellastory.R;
import edu.gatech.spellastory.story_mode.StoryActivity;

public class StoryListAdapter extends RecyclerView.Adapter<StoryListAdapter.ViewHolder> {

    private Context context;
    private List<String> stories;
    private StoryClickListener storyClickListener;

    StoryListAdapter(Context context, List<String> stories) {
        this.context = context;
        this.stories = stories;
    }

    public void setStoryClickListener(StoryClickListener storyClickListener) {
        this.storyClickListener = storyClickListener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        LayoutInflater inflater = LayoutInflater.from(context);
        View view = inflater.inflate(R.layout.item_story_template_list, viewGroup, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, final int i) {
        ImageButton image = viewHolder.imageStory;
        TextView title = viewHolder.storyTitle;

        final String storyTitle = stories.get(i);
        image.setImageDrawable(setPictureFor(storyTitle));
        title.setText(storyTitle);

        image.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                storyClickListener.onStoryClick(storyTitle);
            }
        });
    }

    @Override
    public int getItemCount() {
        return stories.size();
    }

    private Drawable setPictureFor(String word) {
        try {
            InputStream ims = context.getAssets().open("pictures/story_templates/" + word + ".png");
            return Drawable.createFromStream(ims, null);
        } catch (IOException e) {
            // Could not find picture for story template
            e.printStackTrace();
        }
        return null;
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        ImageButton imageStory;
        TextView storyTitle;

        ViewHolder(View itemView) {
            super(itemView);
            imageStory = itemView.findViewById(R.id.iv_story_picture);
            storyTitle = itemView.findViewById(R.id.tx_story_title);
        }
    }

    interface StoryClickListener {
        void onStoryClick(String story);
    }
}

File: app/src/main/java/edu/gatech/spellastory/story/menu/StoryListAdapter.kt
package edu.gatech.spellastory.story.menu

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import edu.gatech.spellastory.R
import edu.gatech.spellastory.domain.stories.Story
import edu.gatech.spellastory.util.getStoryDrawable
import kotlinx.android.synthetic.main.item_story_list.view.*

class StoryListAdapter(private val stories: List<Story>, private val clickListener: (Story) -> Unit) :
        RecyclerView.Adapter<StoryListAdapter.ViewHolder>() {

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun bind(story: Story, clickListener: (Story) -> Unit) {
            itemView.btn_story.text = story.title

            val drawable = itemView.context.getStoryDrawable(story)
            drawable.setBounds(0, 0, 400, 400)
            itemView.btn_story.setCompoundDrawables(drawable, null, null, null)

            itemView.btn_story.setOnClickListener { clickListener(story) }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, i: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_story_list, parent, false))
    }

    override fun getItemCount(): Int {
        return stories.size
    }

    override fun onBindViewHolder(holder: ViewHolder, i: Int) = holder.bind(stories[i], clickListener)
}